statistician.py

```diff
@@ -1,209 +1,114 @@
+          text-align: center;
+          padding: 16px;
+          color: #2870bf;
+        }
+
+        .place {
+          text-align: center;
+        }
+
+        tr:nth-child(even) {
+          background-color: #cccccc;
+        }
+
+        tr:nth-child(odd) {
+          background-color: #ffffff;
+        }
+
+        </style>
         """
-        Private helper method used to update stats related with won and lost with method
-        :param bot1_method: String with method used by BOT1
-        :param bot2_method: String with method used by BOT2
-        :param winner: RoundWinner from enumeration package with information about round winner
+        leaderboard_html = f"""
+        <!DOCTYPE html>
+        <html>
+        <head>
+        {styles}
+        </head>
+        <body>
+        <h1 class="title">Capture Network ELO rating</h1>
+        <div class="center">
+        <table>
+        <tr>
         """
-        bot1 = self.__bank[self.__bot1_name]
-        bot2 = self.__bank[self.__bot2_name]
-        if winner is rw.BOT_1:
-            bot1.won_with_methods[bot1_method] += 1
-            bot2.lost_with_methods[bot2_method] += 1
-        elif winner is rw.BOT_2:
-            bot1.lost_with_methods[bot1_method] += 1
-            bot2.won_with_methods[bot2_method] += 1
-
-        bot1.won_with_methods_per_round.append(copy(bot1.won_with_methods))
-        bot1.lost_with_methods_per_round.append(copy(bot1.lost_with_methods))
-        bot2.won_with_methods_per_round.append(copy(bot2.won_with_methods))
-        bot2.lost_with_methods_per_round.append(copy(bot2.lost_with_methods))
-
-    def __updatePointsEarnedWithMethod(self, bot1_method, bot2_method, winner):
-        """
-        Private helper method used to update stats related with points earned with methods
-        :param bot1_method: String with method used by BOT1
-        :param bot2_method: String with method used by BOT2
-        :param winner: RoundWinner from enumeration package with information about round winner
-        """
-        if winner is rw.BOT_1:
-            points = rules.methodToPrize[rules.nameToMethod[bot1_method]]
-            self.__bank[self.__bot1_name].points_earned_with_method[bot1_method] += points
-        elif winner is rw.BOT_2:
-            points = rules.methodToPrize[rules.nameToMethod[bot2_method]]
-            self.__bank[self.__bot2_name].points_earned_with_method[bot2_method] += points
-
-    def __updateAdvantageAfterRounds(self, advantage):
+        titles = ["Place", "Name", "Fights", "Won", "Lost", "W/L ratio", "ELO"]
+        for title in titles:
+            leaderboard_html += f"""
+            <th>{title}</th>
+            """
+        place = 0
+        bots = db().getBots
+        for row in bots:
+            place += 1
+            data = [place, row[1], row[2] + row[3], row[2], row[3], "----", row[4]]
+            if row[3] + row[2] > 0:
+                data[5] = round(100 * row[2] / (row[3] + row[2]))
```

Comment:
Now you know what row[1], row[2], row[3] means but will you remember this after few months ? :)
It seems to me that the code would be more readable if instead of row[] you used DTO class with proper field names or at least variables with specific names.

Reply:
Answers are in 'titles' list (line 78) ;)
But... ok, i will correct this part.